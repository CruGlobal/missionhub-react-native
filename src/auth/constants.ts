import { AuthPerson_currentUser_person } from './__generated__/AuthPerson';

export enum IdentityProvider {
  Apple,
  Google,
  Facebook,
  TheKey,
  Anonymous,
}

export enum AuthError {
  None,
  CredentialsIncorrect,
  EmailUnverified,
  MfaRequired,
  MfaIncorrect,
  Unknown,
}

export const emptyAuthPerson: AuthPerson_currentUser_person = {
  __typename: 'Person' as const,
  id: '',
  firstName: '',
  lastName: '',
  fullName: '',
  stage: null,
  globalRegistryMdmId: null,
  fbUid: null,
  theKeyUid: null,
};
