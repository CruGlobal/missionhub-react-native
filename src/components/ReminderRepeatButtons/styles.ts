import { StyleSheet } from 'react-native';

import theme from '../../theme';

export default StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'center',
  },
  button: {
    height: 70,
    width: 70,
    margin: 8,
    borderRadius: 35,
    paddingVertical: 0,
    paddingHorizontal: 0,
  },
  buttonInactive: {
    backgroundColor: theme.white,
    borderWidth: 1,
    borderColor: theme.lightGrey,
  },
  buttonActive: {
    backgroundColor: theme.secondaryColor,
  },
  buttonText: {
    ...theme.textRegular12,
    letterSpacing: 1,
  },
  buttonTextInactive: {
    color: theme.lightGrey,
  },
  buttonTextActive: {
    color: theme.white,
  },
});
