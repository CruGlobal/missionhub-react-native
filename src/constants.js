export const LOGIN = 'app/LOGIN';
export const LOGOUT = 'app/LOGOUT';
export const FIRST_TIME = 'app/FIRST_TIME';
export const STAGES = 'app/STAGES';
export const SELECT_STAGE = 'app/SELECT_STAGE';
export const FIRST_NAME_CHANGED = 'app/FIRST_NAME_CHANGED';
export const LAST_NAME_CHANGED = 'app/LAST_NAME_CHANGED';
export const SAVE_NOTES = 'app/SAVE_NOTES';
export const PERSON_FIRST_NAME_CHANGED = 'app/PERSON_FIRST_NAME_CHANGED';
export const PERSON_LAST_NAME_CHANGED = 'app/PERSON_LAST_NAME_CHANGED';
export const ADD_STEP_REMINDER = 'app/ADD_STEP_REMINDER';
export const REMOVE_STEP_REMINDER = 'app/REMOVE_STEP_REMINDER';
export const PUSH_NOTIFICATION_SHOULD_ASK = 'app/PUSH_NOTIFICATION_SHOULD_ASK';
export const PUSH_NOTIFICATION_ASKED = 'app/PUSH_NOTIFICATION_ASKED';
export const PUSH_NOTIFICATION_SET_TOKEN = 'app/PUSH_NOTIFICATION_SET_TOKEN';
export const PUSH_NOTIFICATION_REMINDER = 'app/PUSH_NOTIFICATION_REMINDER';
export const PEOPLE_WITH_ORG_SECTIONS = 'app/PEOPLE_WITH_ORG_SECTIONS';
export const SWIPE_REMINDER_STEPS_HOME = 'app/SWIPE_REMINDER_STEPS_HOME';
export const SWIPE_REMINDER_STEPS_CONTACT = 'app/SWIPE_REMINDER_STEPS_CONTACT';
export const SWIPE_REMINDER_STEPS_REMINDER = 'app/SWIPE_REMINDER_STEPS_REMINDER';
export const SWIPE_REMINDER_JOURNEY = 'app/SWIPE_REMINDER_JOURNEY';
export const SET_VISIBLE_PERSON_INFO = 'app/SET_VISIBLE_PERSON_INFO';
export const UPDATE_VISIBLE_PERSON_INFO = 'app/UPDATE_VISIBLE_PERSON_INFO';
export const SET_JOURNEY_ITEMS = 'app/SET_JOURNEY_ITEMS';
export const CLEAR_JOURNEY_ITEMS = 'app/CLEAR_JOURNEY_ITEMS';
export const COMPLETED_STEP_COUNT = 'app/COMPLETED_STEP_COUNT';
export const UPDATE_STAGES = 'app/UPDATE_STAGES';
export const UPDATE_ONBOARDING_PERSON = 'app/UPDATE_ONBOARDING_PERSON';
export const RESET_ONBOARDING_PERSON = 'app/RESET_ONBOARDING_PERSON';

export const STEP_NOTE = 'stepNote';

export const NAVIGATE_FORWARD = 'Navigation/NAVIGATE';
export const NAVIGATE_RESET = 'Navigation/RESET';
export const MAIN_TABS = 'nav/MAIN_TABS';
export const DRAWER_OPEN = 'DrawerOpen';
export const DRAWER_CLOSE = 'DrawerClose';
export const MAIN_MENU_DRAWER = 'nav/drawer/main';
export const CONTACT_MENU_DRAWER = 'nav/drawer/contact';

export const EXPIRED_ACCESS_TOKEN = 'Expired access token';

export const URL_ENCODED = 'application/x-www-form-urlencoded';
export const THE_KEY_CLIENT_ID = '8480288430352167964';

export const CASEY = 'casey';
export const JEAN = 'jean';

export const LINKS = {
  about: 'https://get.missionhub.com',
  help: 'http://help.missionhub.com',
  playStore: 'market://details?id=com.missionhub',
  appleStore: 'itms://itunes.apple.com/us/app/apple-store/id447869440?mt=8',
  terms: 'https://get.missionhub.com/terms-of-service/',
};

export const ANALYTICS_CONTEXT_CHANGED = 'app/ANALYTICS_CONTEXT_CHANGED';

export const ORG_PERMISSIONS = [ 1, 4 ];

export const ANALYTICS = {
  PAGE_NAME: 'PageName',
  MCID: 'cru.mcid',
  SCREENNAME: 'cru.screenname',
  PREVIOUS_SCREENNAME: 'cru.previousscreenname',
  SITE_SECTION: 'cru.sitesection',
  SITE_SUBSECTION: 'cru.sitesubsection',
  SITE_SUB_SECTION_3: 'cru.subsectionlevel3',
  CONTENT_AUDIENCE_TARGET: 'cru.contentaudiencetarget',
  CONTENT_TOPIC: 'cru.contenttopic',
  LOGGED_IN_STATUS: 'cru.loggedinstatus',
  SSO_GUID: 'cru.ssoguid',
  GR_MASTER_PERSON_ID: 'cru.grmasterpersonid',
  FACEBOOK_ID: 'cru.facebookid',
  CONTENT_LANGUAGE: 'cru.contentlanguage',
};
export const LOGGED_IN = 'logged in';
export const NOT_LOGGED_IN = 'not logged in';

export const INTERACTION_TYPES = {
  MHInteractionTypeAssignedContacts: { id: 100, requestFieldName: 'contact_count', iconName: 'peopleIcon', translationKey: 'interactionAssignedContacts' },
  MHInteractionTypeUncontacted: { id: 101, requestFieldName: 'uncontacted_count', iconName: 'spiritualConversationsIcon', translationKey: 'interactionUncontacted' },
  MHInteractionTypeNote: { id: 1, iconName: 'commentIcon', translationKey: 'interactionNote', hideReport: true },
  MHInteractionTypeSpiritualConversation: { id: 2, iconName: 'spiritualConversationsIcon', translationKey: 'interactionSpiritualConversation' },
  MHInteractionTypeGospelPresentation: { id: 3, iconName: 'gospelIcon', translationKey: 'interactionGospel' },
  MHInteractionTypePersonalDecision: { id: 4, iconName: 'decisionIcon', translationKey: 'interactionDecision' },
  MHInteractionTypeHolySpiritConversation: { id: 5, iconName: 'spiritIcon', translationKey: 'interactionSpirit' },
  // MHInteractionTypeGraduatingOnMission: { id: 6, iconName: '', translationKey: 'interactionNote' },
  MHInteractionTypeDiscipleshipConversation: { id: 9, iconName: 'discipleshipConversationIcon', translationKey: 'interactionDiscipleshipConversation' },
};

export default {

};
