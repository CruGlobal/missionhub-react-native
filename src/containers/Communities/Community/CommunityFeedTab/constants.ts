export const COMMUNITY_FEED = 'nav/COMMUNITY_FEED';
