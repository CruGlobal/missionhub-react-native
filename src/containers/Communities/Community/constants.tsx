import React from 'react';
import i18next from 'i18next';

import GroupChallenges, {
  COMMUNITY_CHALLENGES,
} from '../../Groups/GroupChallenges';

import {
  CommunityFeedTab,
  COMMUNITY_FEED,
} from './CommunityFeedTab/CommunityFeedTab';
import {
  COMMUNITY_IMPACT,
  CommunityImpactTab,
} from './CommunityImpactTab/CommunityImpactTab';
import { CommunitiesCollapsibleHeaderContext } from './CommunityHeader/CommunityHeader';

// Moved here to prevent circular dependency
export const COMMUNITY_TABS = 'nav/COMMUNITY_TABS';

export const communityTabs = [
  {
    name: i18next.t('communityTabs:feed'),
    navigationAction: COMMUNITY_FEED,
    component: () => (
      <CommunityFeedTab
        collapsibleHeaderContext={CommunitiesCollapsibleHeaderContext}
      />
    ),
  },
  {
    name: i18next.t('communityTabs:challenges'),
    navigationAction: COMMUNITY_CHALLENGES,
    component: GroupChallenges,
  },
  {
    name: i18next.t('communityTabs:impact'),
    navigationAction: COMMUNITY_IMPACT,
    component: () => (
      <CommunityImpactTab
        collapsibleHeaderContext={CommunitiesCollapsibleHeaderContext}
      />
    ),
  },
];
