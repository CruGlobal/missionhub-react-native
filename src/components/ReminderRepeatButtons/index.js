import React, { Component } from 'react';
import { View } from 'react-native';
import { translate } from 'react-i18next';

import { Button } from '../common';

import styles from './styles';

@translate('stepReminder')
export default class ReminderRepeatButtons extends Component {
  state = {
    dailyActive: false,
    weeklyActive: false,
    monthlyActive: false,
  };

  setButtonState(dailyActive, weeklyActive, monthlyActive) {
    this.setState({ dailyActive, weeklyActive, monthlyActive });
  }

  handleSetDaily = () => {
    this.setButtonState(!this.state.dailyActive, false, false);
  };

  handleSetWeekly = () => {
    this.setButtonState(false, !this.state.weeklyActive, false);
  };

  handleSetMonthly = () => {
    this.setButtonState(false, false, !this.state.monthlyActive);
  };

  render() {
    const { t } = this.props;
    const { dailyActive, weeklyActive, monthlyActive } = this.state;
    const {
      container,
      button,
      buttonInactive,
      buttonActive,
      buttonText,
      buttonTextInactive,
      buttonTextActive,
    } = styles;

    return (
      <View style={container}>
        <Button
          style={[button, dailyActive ? buttonActive : buttonInactive]}
          buttonTextStyle={[
            buttonText,
            dailyActive ? buttonTextActive : buttonTextInactive,
          ]}
          text={t('daily')}
          onPress={this.handleSetDaily}
        />
        <Button
          style={[button, weeklyActive ? buttonActive : buttonInactive]}
          buttonTextStyle={[
            buttonText,
            weeklyActive ? buttonTextActive : buttonTextInactive,
          ]}
          text={t('weekly')}
          onPress={this.handleSetWeekly}
        />
        <Button
          style={[button, monthlyActive ? buttonActive : buttonInactive]}
          buttonTextStyle={[
            buttonText,
            monthlyActive ? buttonTextActive : buttonTextInactive,
          ]}
          text={t('monthly')}
          onPress={this.handleSetMonthly}
        />
      </View>
    );
  }
}
