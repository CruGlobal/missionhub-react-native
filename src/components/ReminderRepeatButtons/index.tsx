import React from 'react';
import { View } from 'react-native';
import { useTranslation } from 'react-i18next';

import { Button } from '../common';
import {
  REMINDER_RECURRENCES,
  REMINDER_RECURRENCES_ENUM,
} from '../../constants';

import styles from './styles';

const { ONCE, DAILY, WEEKLY, MONTHLY } = REMINDER_RECURRENCES;

interface ReminderRepeatButtonsProps {
  recurrence?: REMINDER_RECURRENCES_ENUM;
  onRecurrenceChange: Function;
}

const ReminderRepeatButtons = ({
  recurrence: currentRecurrence,
  onRecurrenceChange,
}: ReminderRepeatButtonsProps) => {
  const { t } = useTranslation('stepReminder');
  function setButtonState(clickedRecurrence: REMINDER_RECURRENCES_ENUM) {
    const recurrence =
      currentRecurrence === clickedRecurrence ? ONCE : clickedRecurrence;
    onRecurrenceChange(recurrence);
  }
  const handleSetDaily = () => setButtonState(DAILY);
  const handleSetWeekly = () => setButtonState(WEEKLY);
  const handleSetMonthly = () => setButtonState(MONTHLY);
  function renderReminderButton(
    recurrence: REMINDER_RECURRENCES_ENUM,
    onPress: Function,
  ) {
    const {
      button,
      buttonInactive,
      buttonActive,
      buttonText,
      buttonTextInactive,
      buttonTextActive,
    } = styles;
    const active = currentRecurrence === recurrence;

    return (
      <Button
        testID={
          recurrence === DAILY
            ? 'ReminderRepeatButtonDaily'
            : recurrence === WEEKLY
            ? 'ReminderRepeatButtonWeekly'
            : 'ReminderRepeatButtonMonthly'
        }
        style={[button, active ? buttonActive : buttonInactive]}
        buttonTextStyle={[
          buttonText,
          active ? buttonTextActive : buttonTextInactive,
        ]}
        text={t(recurrence)}
        onPress={onPress}
      />
    );
  }
  const { container } = styles;
  return (
    <View style={container}>
      {renderReminderButton(DAILY, handleSetDaily)}
      {renderReminderButton(WEEKLY, handleSetWeekly)}
      {renderReminderButton(MONTHLY, handleSetMonthly)}
    </View>
  );
};

export default ReminderRepeatButtons;
