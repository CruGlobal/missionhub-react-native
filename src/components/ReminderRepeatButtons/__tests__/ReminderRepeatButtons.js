import React from 'react';

import ReminderRepeatButtons from '..';

import { renderShallow } from '../../../../testUtils';
import { REMINDER_RECURRENCES } from '../../../constants';

const { ONCE, DAILY, WEEKLY, MONTHLY } = REMINDER_RECURRENCES;

const onRecurrenceChange = jest.fn();

let component;

const test = () => expect(component).toMatchSnapshot();

beforeEach(() => {
  onRecurrenceChange.mockClear();
  component = renderShallow(
    <ReminderRepeatButtons onRecurrenceChange={onRecurrenceChange} />,
  );
});

describe('none selected', () => {
  it('renders with none highlighted', () => {
    test();
  });

  it('calls onRecurrenceChange with ONCE', () => {
    expect(onRecurrenceChange).toHaveBeenCalledTimes(1);
    expect(onRecurrenceChange).toHaveBeenCalledWith(ONCE);
  });
});

describe('starts with weekly selected', () => {
  beforeEach(() => {
    onRecurrenceChange.mockClear();
    component = renderShallow(
      <ReminderRepeatButtons
        recurrence={WEEKLY}
        onRecurrenceChange={onRecurrenceChange}
      />,
    );
  });

  it('renders with weekly highlighted', () => {
    test();
  });

  it('calls onRecurrenceChange with ONCE', () => {
    expect(onRecurrenceChange).toHaveBeenCalledTimes(1);
    expect(onRecurrenceChange).toHaveBeenCalledWith(WEEKLY);
  });
});

describe('select daily button', () => {
  beforeEach(() => {
    component
      .childAt(0)
      .props()
      .onPress();
    component.update();
  });

  it('calls onRecurrenceChange with DAILY', () => {
    expect(onRecurrenceChange).toHaveBeenCalledTimes(2);
    expect(onRecurrenceChange).toHaveBeenCalledWith(DAILY);
  });

  describe('select button again', () => {
    beforeEach(() => {
      component
        .childAt(0)
        .props()
        .onPress();
      component.update();
    });

    it('calls onRecurrenceChange with ONCE', () => {
      expect(onRecurrenceChange).toHaveBeenCalledTimes(3);
      expect(onRecurrenceChange).toHaveBeenCalledWith(ONCE);
    });
  });

  describe('select other button', () => {
    beforeEach(() => {
      component
        .childAt(1)
        .props()
        .onPress();
      component.update();
    });

    it('calls onRecurrenceChange with WEEKLY', () => {
      expect(onRecurrenceChange).toHaveBeenCalledTimes(3);
      expect(onRecurrenceChange).toHaveBeenCalledWith(WEEKLY);
    });
  });
});

describe('select weekly button', () => {
  beforeEach(() => {
    component
      .childAt(1)
      .props()
      .onPress();
    component.update();
  });

  it('calls onRecurrenceChange with WEEKLY', () => {
    expect(onRecurrenceChange).toHaveBeenCalledTimes(2);
    expect(onRecurrenceChange).toHaveBeenCalledWith(WEEKLY);
  });
});

describe('select monthly button', () => {
  beforeEach(() => {
    component
      .childAt(2)
      .props()
      .onPress();
    component.update();
  });

  it('calls onRecurrenceChange with MONTHLY', () => {
    expect(onRecurrenceChange).toHaveBeenCalledTimes(2);
    expect(onRecurrenceChange).toHaveBeenCalledWith(MONTHLY);
  });
});
