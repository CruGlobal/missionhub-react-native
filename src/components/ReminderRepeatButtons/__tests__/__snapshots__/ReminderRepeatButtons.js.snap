// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`none selected renders with none highlighted 1`] = `
<Component
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <Button
    buttonTextStyle={
      Array [
        Object {
          "fontSize": 12,
          "fontWeight": "normal",
          "letterSpacing": 1,
          "lineHeight": 16,
        },
        Object {
          "color": "#B4B6BA",
        },
      ]
    }
    onPress={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 35,
          "height": 70,
          "margin": 8,
          "paddingHorizontal": 0,
          "paddingVertical": 0,
          "width": 70,
        },
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#B4B6BA",
          "borderWidth": 1,
        },
      ]
    }
    text="Daily"
  />
  <Button
    buttonTextStyle={
      Array [
        Object {
          "fontSize": 12,
          "fontWeight": "normal",
          "letterSpacing": 1,
          "lineHeight": 16,
        },
        Object {
          "color": "#B4B6BA",
        },
      ]
    }
    onPress={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 35,
          "height": 70,
          "margin": 8,
          "paddingHorizontal": 0,
          "paddingVertical": 0,
          "width": 70,
        },
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#B4B6BA",
          "borderWidth": 1,
        },
      ]
    }
    text="Weekly"
  />
  <Button
    buttonTextStyle={
      Array [
        Object {
          "fontSize": 12,
          "fontWeight": "normal",
          "letterSpacing": 1,
          "lineHeight": 16,
        },
        Object {
          "color": "#B4B6BA",
        },
      ]
    }
    onPress={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 35,
          "height": 70,
          "margin": 8,
          "paddingHorizontal": 0,
          "paddingVertical": 0,
          "width": 70,
        },
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#B4B6BA",
          "borderWidth": 1,
        },
      ]
    }
    text="Monthly"
  />
</Component>
`;

exports[`select daily button renders with daily button highlighted only 1`] = `
<Component
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <Button
    buttonTextStyle={
      Array [
        Object {
          "fontSize": 12,
          "fontWeight": "normal",
          "letterSpacing": 1,
          "lineHeight": 16,
        },
        Object {
          "color": "#ffffff",
        },
      ]
    }
    onPress={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 35,
          "height": 70,
          "margin": 8,
          "paddingHorizontal": 0,
          "paddingVertical": 0,
          "width": 70,
        },
        Object {
          "backgroundColor": "#52C5DC",
        },
      ]
    }
    text="Daily"
  />
  <Button
    buttonTextStyle={
      Array [
        Object {
          "fontSize": 12,
          "fontWeight": "normal",
          "letterSpacing": 1,
          "lineHeight": 16,
        },
        Object {
          "color": "#B4B6BA",
        },
      ]
    }
    onPress={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 35,
          "height": 70,
          "margin": 8,
          "paddingHorizontal": 0,
          "paddingVertical": 0,
          "width": 70,
        },
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#B4B6BA",
          "borderWidth": 1,
        },
      ]
    }
    text="Weekly"
  />
  <Button
    buttonTextStyle={
      Array [
        Object {
          "fontSize": 12,
          "fontWeight": "normal",
          "letterSpacing": 1,
          "lineHeight": 16,
        },
        Object {
          "color": "#B4B6BA",
        },
      ]
    }
    onPress={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 35,
          "height": 70,
          "margin": 8,
          "paddingHorizontal": 0,
          "paddingVertical": 0,
          "width": 70,
        },
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#B4B6BA",
          "borderWidth": 1,
        },
      ]
    }
    text="Monthly"
  />
</Component>
`;

exports[`select daily button select button again renders with none highlighted 1`] = `
<Component
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <Button
    buttonTextStyle={
      Array [
        Object {
          "fontSize": 12,
          "fontWeight": "normal",
          "letterSpacing": 1,
          "lineHeight": 16,
        },
        Object {
          "color": "#B4B6BA",
        },
      ]
    }
    onPress={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 35,
          "height": 70,
          "margin": 8,
          "paddingHorizontal": 0,
          "paddingVertical": 0,
          "width": 70,
        },
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#B4B6BA",
          "borderWidth": 1,
        },
      ]
    }
    text="Daily"
  />
  <Button
    buttonTextStyle={
      Array [
        Object {
          "fontSize": 12,
          "fontWeight": "normal",
          "letterSpacing": 1,
          "lineHeight": 16,
        },
        Object {
          "color": "#B4B6BA",
        },
      ]
    }
    onPress={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 35,
          "height": 70,
          "margin": 8,
          "paddingHorizontal": 0,
          "paddingVertical": 0,
          "width": 70,
        },
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#B4B6BA",
          "borderWidth": 1,
        },
      ]
    }
    text="Weekly"
  />
  <Button
    buttonTextStyle={
      Array [
        Object {
          "fontSize": 12,
          "fontWeight": "normal",
          "letterSpacing": 1,
          "lineHeight": 16,
        },
        Object {
          "color": "#B4B6BA",
        },
      ]
    }
    onPress={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 35,
          "height": 70,
          "margin": 8,
          "paddingHorizontal": 0,
          "paddingVertical": 0,
          "width": 70,
        },
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#B4B6BA",
          "borderWidth": 1,
        },
      ]
    }
    text="Monthly"
  />
</Component>
`;

exports[`select daily button select other button renders with other highlighted only 1`] = `
<Component
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <Button
    buttonTextStyle={
      Array [
        Object {
          "fontSize": 12,
          "fontWeight": "normal",
          "letterSpacing": 1,
          "lineHeight": 16,
        },
        Object {
          "color": "#B4B6BA",
        },
      ]
    }
    onPress={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 35,
          "height": 70,
          "margin": 8,
          "paddingHorizontal": 0,
          "paddingVertical": 0,
          "width": 70,
        },
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#B4B6BA",
          "borderWidth": 1,
        },
      ]
    }
    text="Daily"
  />
  <Button
    buttonTextStyle={
      Array [
        Object {
          "fontSize": 12,
          "fontWeight": "normal",
          "letterSpacing": 1,
          "lineHeight": 16,
        },
        Object {
          "color": "#ffffff",
        },
      ]
    }
    onPress={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 35,
          "height": 70,
          "margin": 8,
          "paddingHorizontal": 0,
          "paddingVertical": 0,
          "width": 70,
        },
        Object {
          "backgroundColor": "#52C5DC",
        },
      ]
    }
    text="Weekly"
  />
  <Button
    buttonTextStyle={
      Array [
        Object {
          "fontSize": 12,
          "fontWeight": "normal",
          "letterSpacing": 1,
          "lineHeight": 16,
        },
        Object {
          "color": "#B4B6BA",
        },
      ]
    }
    onPress={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 35,
          "height": 70,
          "margin": 8,
          "paddingHorizontal": 0,
          "paddingVertical": 0,
          "width": 70,
        },
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#B4B6BA",
          "borderWidth": 1,
        },
      ]
    }
    text="Monthly"
  />
</Component>
`;

exports[`select monthly button renders with monthly button highlighted only 1`] = `
<Component
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <Button
    buttonTextStyle={
      Array [
        Object {
          "fontSize": 12,
          "fontWeight": "normal",
          "letterSpacing": 1,
          "lineHeight": 16,
        },
        Object {
          "color": "#B4B6BA",
        },
      ]
    }
    onPress={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 35,
          "height": 70,
          "margin": 8,
          "paddingHorizontal": 0,
          "paddingVertical": 0,
          "width": 70,
        },
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#B4B6BA",
          "borderWidth": 1,
        },
      ]
    }
    text="Daily"
  />
  <Button
    buttonTextStyle={
      Array [
        Object {
          "fontSize": 12,
          "fontWeight": "normal",
          "letterSpacing": 1,
          "lineHeight": 16,
        },
        Object {
          "color": "#B4B6BA",
        },
      ]
    }
    onPress={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 35,
          "height": 70,
          "margin": 8,
          "paddingHorizontal": 0,
          "paddingVertical": 0,
          "width": 70,
        },
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#B4B6BA",
          "borderWidth": 1,
        },
      ]
    }
    text="Weekly"
  />
  <Button
    buttonTextStyle={
      Array [
        Object {
          "fontSize": 12,
          "fontWeight": "normal",
          "letterSpacing": 1,
          "lineHeight": 16,
        },
        Object {
          "color": "#ffffff",
        },
      ]
    }
    onPress={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 35,
          "height": 70,
          "margin": 8,
          "paddingHorizontal": 0,
          "paddingVertical": 0,
          "width": 70,
        },
        Object {
          "backgroundColor": "#52C5DC",
        },
      ]
    }
    text="Monthly"
  />
</Component>
`;

exports[`select weekly button renders with weekly button highlighted only 1`] = `
<Component
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <Button
    buttonTextStyle={
      Array [
        Object {
          "fontSize": 12,
          "fontWeight": "normal",
          "letterSpacing": 1,
          "lineHeight": 16,
        },
        Object {
          "color": "#B4B6BA",
        },
      ]
    }
    onPress={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 35,
          "height": 70,
          "margin": 8,
          "paddingHorizontal": 0,
          "paddingVertical": 0,
          "width": 70,
        },
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#B4B6BA",
          "borderWidth": 1,
        },
      ]
    }
    text="Daily"
  />
  <Button
    buttonTextStyle={
      Array [
        Object {
          "fontSize": 12,
          "fontWeight": "normal",
          "letterSpacing": 1,
          "lineHeight": 16,
        },
        Object {
          "color": "#ffffff",
        },
      ]
    }
    onPress={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 35,
          "height": 70,
          "margin": 8,
          "paddingHorizontal": 0,
          "paddingVertical": 0,
          "width": 70,
        },
        Object {
          "backgroundColor": "#52C5DC",
        },
      ]
    }
    text="Weekly"
  />
  <Button
    buttonTextStyle={
      Array [
        Object {
          "fontSize": 12,
          "fontWeight": "normal",
          "letterSpacing": 1,
          "lineHeight": 16,
        },
        Object {
          "color": "#B4B6BA",
        },
      ]
    }
    onPress={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 35,
          "height": 70,
          "margin": 8,
          "paddingHorizontal": 0,
          "paddingVertical": 0,
          "width": 70,
        },
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#B4B6BA",
          "borderWidth": 1,
        },
      ]
    }
    text="Monthly"
  />
</Component>
`;
